### PR TITLE
Pointer attributes in C

### PR DESCRIPTION
**Pull request by ntamas**
This PR adds support for storing pointers as attribute values in C, in response to #2360.

Still a work in progress, no unit tests yet, I just added the new bits of code by copying similar handlers for numeric attributes and changing the data types.

### CODE REVIEW

**Comment by szhorvat** [comment]
What's the approach here? Will we have custom constructors and destructors?  Or just plain pointers with null pointers also allowed, leaving memory management to the user?

Suppose a vertex is deleted. How is the pointed-to object deallocated?  What happens when adding a new vertex?  How is the object created?

File: src/graph/cattributes.c [hunk 1]
```diff
@@ -537,6 +591,11 @@ static void igraph_i_attribute_permutation_work_area_destroy(
         igraph_Free(work_area->boolean);
         work_area->boolean = NULL;
     }
+    if (work_area->pointers != NULL) {
+        igraph_vector_ptr_destroy(work_area->pointers);
+        igraph_Free(work_area->pointers);
```
Comment (szhorvat) [inline src/graph/cattributes.c:596]: This should be `IGRAPH_FREE`

**Comment by szhorvat** [comment]
I see that the approach was not to take ownership of the pointed to object.

But this is only really useful for static graphs. For dynamically changing graphs it is very limited. For a static graph, one might as well use a plain vector of objects without resorting to attributes. The true value of attributes is that they get automatically managed when the graph changes.

**Comment by szhorvat** [comment]
@ntamas What do we do with this? Are interested in pushing it further for 1.0?

**Comment by ntamas** [comment]
No, not really. I have plenty of other things on my plate.

**Pull request closed without merging**